src/telemetry/Manager.cc

```diff
@@ -259,6 +259,32 @@ static bool compare_histograms(const std::optional<ZVal>& a, const std::optional
     return comparer(a, b, metric_record_type);
 }
 
+static VectorValPtr build_label_values_vector(const std::vector<prometheus::ClientMetric::Label>& prom_labels,
+                                              const VectorValPtr& record_label_names) {
+    static auto string_vec_type = zeek::id::find_type<zeek::VectorType>("string_vec");
+    auto label_values_vec = make_intrusive<VectorVal>(string_vec_type);
+
+    // This feels really bad, since it's an O(m*n) search to bulld the vector,
```

Comment:
If we change the label storage in the `Metric` record to a `table[string] of string` we'd avoid this scan, no? I'm worried that the separation of labels into two vectors, one in `MetricOpts` and one in `Metric`, is always going to be tricky for the user.

Reply:
No, because this is scanning through the `vector<pair<string,string>>` that prometheus-cpp returns from the call to `Collect()`. If they'd return a map instead, it would be cleaner.

Reply:
That said, we've already forked prometheus-cpp to add the async interface, so modifying it to return a map instead shouldn't be too hard.

Reply:
This was superceded by the change that moves the labels to the other records, so I'm going to resolve this comment.